src/vs/vscode.proposed.d.ts

```diff
@@ -23,6 +23,31 @@ declare module 'vscode' {
 	export namespace window {
 		export function showOpenDialog(options: OpenDialogOptions): Thenable<Uri[]>;
 		export function showSaveDialog(options: SaveDialogOptions): Thenable<Uri>;
+
+		/**
+		 * Shows a selection list of [workspace folders](#workspace.workspaceFolders) to pick from.
+		 * Returns `undefined` if no folder is open.
+		 *
+		 * @param options Configures the behavior of the workspace folder list.
+		 * @return A promise that resolves to the workspace folder or `undefined`.
+		 */
+		export function showWorkspaceFolderPick(options?: WorkspaceFolderPickOptions): Thenable<WorkspaceFolder | undefined>;
+	}
+
+	/**
+	 * Options to configure the behaviour of the [workspace folder](#WorkspaceFolder) pick UI.
+	 */
+	export interface WorkspaceFolderPickOptions {
```

Comment:
Don't define types inside a namespace but globally

Reply:
@jrieken isn't that globally? It is outside the `window` namespace.

Reply:
Oh, sorry. Didn't see that closing-curly